planet/management/commands/update_planet.py

```diff
@@ -0,0 +1,139 @@
+"""
+update_planet
+
+Imports all feeds for users who have filled in a valid website and website_rss,
+the amount of items imported is limited to the defined FEED_LIMT
+
+Usage: ./manage.py update_planet
+"""
+
+
+import logging
+import sys
+import time
+
+from datetime import datetime
+from pytz import utc
+
+import bleach
+import feedparser
+
+from django.core.cache import cache
+from django.core.management.base import BaseCommand
+from django.template.defaultfilters import truncatewords_html
+from django.conf import settings
+
+from planet.models import Feed, FeedItem, FEEDITEM_SUMMARY_LIMIT
+
+
+logging.basicConfig(
+    level=logging.WARNING,
+    format=u'%(asctime)s -> %(levelname)s: %(message)s',
+    datefmt=u'%Y-%m-%d %H:%M:%S',
+    stream=sys.stderr)
+logger = logging.getLogger()
+
+
+class Command(BaseCommand):
+
+    def parse_feed(self, feedmodel):
+        latest = None
+        feeds = []
+        url = feedmodel.website_rss
+
+        logger.debug("Import new feed items for '%s'", url)
+
+        try:
+            latest = FeedItem.objects.filter(feed=feedmodel).latest()
+        except FeedItem.DoesNotExist:
+            pass
+
+        etag = cache.get(url)
+        feed = feedparser.parse(url, etag=etag)
+
+        if feed['status'] == 304:
+            logger.info("The feed '%s' has not changed since we last checked it", url)
+
+        if feed['status'] != 200:
+            logger.info("error parsing feed: '%s', status: '%s'", url, feed['status'])
+            return feeds
+
+        if not feed.entries:
+            logger.info("error parsing feed: '%s', feed has no entries", url)
+            return feeds
+
+        for entry in feed.entries[:settings.RSS_FEED_LIMIT]:
+            item = self.parse_entry(entry, latest, url)
+
+            if not item:
+                continue
+
+            item.feed = feedmodel
+            feeds.append(item)
+
+        # TODO: only set cache when sucesfully changed?
```

Comment:
It would probably be better to move the `bulk_create` to just above this `cache.set`, yes.